libs/wingc/src/type_check.rs

```diff
@@ -1590,6 +1602,51 @@ impl<'a> TypeChecker<'a> {
 		}
 	}
 
+	fn type_check_loop(
+		&mut self,
+		iterator: &Symbol,
+		expr: &Expr,
+		statements: &Scope,
+		stmt: &Stmt,
+		env: &mut SymbolEnv,
+		context: &TypeCheckerContext,
+	) {
+		// TODO: Expression must be iterable
+		let exp_type = self.type_check_exp(expr, env, stmt.idx, context);
+
+		let iterator_type = match &*exp_type {
+			// These are builtin iterables that have a clear/direct iterable type
+			Type::Array(t) => *t,
+			Type::Set(t) => *t,
+			Type::MutArray(t) => *t,
+			Type::MutSet(t) => *t,
+
+			// TODO: Handle non-builtin iterables
+			t => {
+				self.type_error(TypeError {
+					message: format!("Unable to iterate over \"{}\"", t),
+					span: expr.span.clone(),
+				});
+				self.types.anything()
+			}
+		};
```

Comment:
Let's extract this into a function like `exp_type.is_iterable()`, which returns `true` for collections and `NumberSequence` and `false` for everything else

Reply:
I created the is_iterable() function, but I need this variable iterator_type to get the type of variable received by the array.